whisker/docker-image/Dockerfile.nginx

```diff
@@ -12,20 +12,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+ARG QEMU_IMAGE
+
+FROM ${QEMU_IMAGE} as qemu
+
 FROM registry.access.redhat.com/ubi8/ubi:latest as ubi
 
+COPY --from=qemu /usr/bin/qemu-*-static /usr/bin/
+
 RUN dnf upgrade -y
 
 COPY docker-image/nginx.repo /etc/yum.repos.d/nginx.repo
 
 RUN dnf --enablerepo=nginx-stable install -y \
     nginx
 
+FROM scratch as source
+
+# binary and dependencies
+COPY --from=ubi /usr/sbin/nginx /usr/sbin/nginx
+
+COPY --from=ubi /bin/sh /bin/sh
+COPY --from=ubi /usr/bin/cat /usr/bin/cat
+COPY --from=ubi /usr/bin/cd /usr/bin/cd
+COPY --from=ubi /usr/bin/coreutils /usr/bin/coreutils
+COPY --from=ubi /usr/bin/sed /usr/bin/sed
+
+COPY --from=ubi /lib64/libacl.so.1 /lib64/libacl.so.1
+COPY --from=ubi /lib64/libattr.so.1 /lib64/libattr.so.1
+COPY --from=ubi /lib64/libcap.so.2 /lib64/libcap.so.2
+COPY --from=ubi /lib64/libcrypt.so.1 /lib64/libcrypt.so.1
+COPY --from=ubi /lib64/libcrypto.so.1.1 /lib64/libcrypto.so.1.1
+COPY --from=ubi /lib64/libdl.so.2 /lib64/libdl.so.2
+COPY --from=ubi /lib64/libpcre2-8.so.0 /lib64/libpcre2-8.so.0
+COPY --from=ubi /lib64/librt.so.1 /lib64/librt.so.1
+COPY --from=ubi /lib64/libselinux.so.1 /lib64/libselinux.so.1
+COPY --from=ubi /lib64/libssl.so.1.1 /lib64/libssl.so.1.1
+COPY --from=ubi /lib64/libtinfo.so.6 /lib64/libtinfo.so.6
+COPY --from=ubi /lib64/libz.so.1 /lib64/libz.so.1
+
+# web server configs and folder permissions
+COPY --from=ubi --chown=10001:10001 /etc/nginx /etc/nginx/
+# nginx-start.sh writes clientsideConfig.js into this folder
+COPY --from=ubi --chown=10001:10001 /usr/share/nginx/html /usr/share/nginx/html/
+COPY --from=ubi --chown=10001:10001 /var/cache/nginx /var/cache/nginx/
+COPY --from=ubi --chown=10001:10001 /var/log/nginx /var/log/nginx/
+
 COPY docker-image/nginx-start.sh /usr/bin/nginx-start.sh
 
 COPY docker-image/default.conf /etc/nginx/conf.d/default.conf
 COPY docker-image/nginx.conf /etc/nginx/nginx.conf
 
 COPY dist /usr/share/nginx/html/
 
+FROM calico/base
```

Comment:
Use `${CALICO_BASE}` arg like other components. I am going to [pin](https://github.com/projectcalico/calico/pull/9867) `calico/base` to a release tag.

Reply:
Done.